src/utils/mod.rs

```diff
@@ -108,29 +109,26 @@ fn get_allowed_filesystems(top_level_names: &HashSet<String>) -> Option<HashSet<
     Some(limit_filesystems)
 }
 
-pub fn normalize_path<P: AsRef<std::path::Path>>(path: P) -> std::string::String {
+pub fn normalize_path<P: AsRef<Path>>(path: P) -> PathBuf {
     // normalize path ...
     // 1. removing repeated separators
     // 2. removing interior '.' ("current directory") path segments
     // 3. removing trailing extra separators and '.' ("current directory") path segments
     // * `Path.components()` does all the above work; ref: <https://doc.rust-lang.org/std/path/struct.Path.html#method.components>
     // 4. changing to os preferred separator (automatically done by recollecting components back into a PathBuf)
-    path.as_ref()
-        .components()
-        .collect::<std::path::PathBuf>()
-        .to_string_lossy()
-        .to_string()
+    path.as_ref().components().collect::<PathBuf>()
```

Comment:
do you think we can use canonicalize? https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.canonicalize

Reply:
Yes that's a possible alternative, but this method always succeeds. `fs::canonicalize()`, on the other hand, requires an existing path or returns an error which would then have to be handled.

Reply:
good reasoning, agreed